src/engine/source/hlp/src/SpecificParsers.cpp

```diff
@@ -376,3 +378,113 @@ bool parseURL(const char **it, char endToken, URLResult &result) {
 
     return true;
 }
+
+bool parseDomain(const char **it, char endToken, std::vector<std::string> const& captureOpts, DomainResult &result) {
+    static const std::string_view valid_chars {"abcdefghijklmnopqrstuvwxyz"
+                                               "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+                                               "0123456789-_."};
+    static const int DOMAIN_MAX_SIZE = 253;
+    static const int LABEL_MAX_SIZE = 63;
+
+    const char *start = *it;
+    while (**it != endToken || **it != '\0') {
+        (*it)++;
+    }
+    std::string_view str { start, (size_t)((*it) - start) };
+
+    size_t protocol_end = str.find("://");
+    size_t domain_start = 0;
+    // Protocol
+    if (std::string::npos != protocol_end) {
+        result.protocol = str.substr(0, protocol_end);
+        domain_start = protocol_end + 3;
+    }
+    size_t domain_end = str.find("/", protocol_end + 3);
+    // Domain
+    result.domain = str.substr(domain_start, domain_end - domain_start);
+    // Domain length check
+    if (result.domain.empty() || result.domain.length() > DOMAIN_MAX_SIZE){
+        //TODO Log domain size error
+        return false;
+    }
+    // Domain valid characters check
+    for (char const &c: result.domain) {
+        if (std::string::npos == valid_chars.find(c)) {
+            //TODO Log invalid char error
+            return false;
+        }
+    }
```

Comment:
to avoid the linear complexity this could be 
```suggestion
    // Domain valid characters check
    for (char const &c: result.domain) {
        if (!std::isalnum(c) && c != '-' && c != '_' && c != '.') {
            //TODO Log invalid char error
            return false;
        }
    }
```

We could also avoid using the std::isalnum which involves the locale by just doing
```suggestion
bool isAsciiNum(char c){ return (c <= 57 && c >= 48);}
bool isAsciiUpp(char c){ return (c <= 90 && c >= 65);}
bool isAsciiLow(char c){ return (c <= 122 && c >= 97);}

    // Domain valid characters check
    for (char const &c: result.domain) {
        if (!isAsciiNum(c) && !isAsciiUpp(c) && !isAsciiLow(c) &&  c != '-' && c != '_' && c != '.') {
            return false;
        }
    }
```

Reply:
Changed!